### DISCUSSION TITLE
insert_text insert_textbox and annotation's insert_free_text both hit a space, and the next word goes out of the area and breaks the line, but Chinese has no spaces, which leads to a single space, and then a sentence following it is treated as a word.

### DISCUSSION THREAD

**Discussion by izerui** [pymupdf/PyMuPDF#3372] [Q&A] [answered]
### Description of the bug

sample:
世界 人民大团结万岁!

result:
line1: 世界
line2: 人民大团结万岁!

### How to reproduce the bug

What is actually expected is:
line1: 世界 人民大
line2: 团结万岁!

### PyMuPDF version

1.24.1

### Operating system

MacOS

### Python version

3.10

**Comment by izerui** [comment]
Whether to consider an additional parameter or option, whether to hit a space for a newline

**Comment by JorjMcKie** [comment]
This is no issue, but a Discussionsitem.

**Reply by izerui** [reply]
Because English words are not separable, whereas CJK individual characters are separable

**Comment by JorjMcKie** [comment, marked as answer]
Well, why do you say the separation should happen as you indicate? There either is a space in the text or not.
If there is, then it will be used for line breaks when necessary.
Otherwise, the text is treated like a word with a length exceeding the rectangle width and it will be split as required at an arbitrary point.
There is no way to avoid this.

**_You_** must become active here:

1. If the comfort coming with `insert_textbox` is not usable for, then use `insert_text` instead, based on your knowing better when to start a new line. Use `Font.tength_length` to find the length of a given string. As Chinese text is monospaced, you should easily be able to compute the right spots where to start a new line.
2. Or:
   - use `page.insert_htmlbox()` instead
   - prepare your text with soft hyphen characters where you would like to have them. This is character `&shy`;. It will stay invisible if not needed, otherwise, a "-" will be generated at line ends.

**Reply by izerui** [reply]
Sorry, maybe my expression is not too clear, the space in Chinese is also counted as a character, may be in English as a spacing between words, when a paragraph of text that contains both Chinese and English, this time to meet both can not be split into English words, nor should a string of Chinese as a word to directly line feed, maybe this scene in some countries will be more common, it's okay, I think I It seems that it can be solved by inserting html rich text, he perfectly adapts to line breaks.
